src/GSoft.Extensions.Mongo.ApplicationInsights/ApplicationInsightsEventSubscriberFactory.cs

```diff
@@ -0,0 +1,26 @@
+﻿using Microsoft.ApplicationInsights;
+using Microsoft.Extensions.Options;
+using MongoDB.Driver.Core.Events;
+
+namespace GSoft.Extensions.Mongo.ApplicationInsights;
+
+internal sealed class ApplicationInsightsEventSubscriberFactory : IMongoEventSubscriberFactory
+{
+    private readonly IOptionsMonitor<MongoClientOptions> _optionsMonitor;
+    private readonly TelemetryClient? _telemetryClient;
+
+    public ApplicationInsightsEventSubscriberFactory(IOptionsMonitor<MongoClientOptions> optionsMonitor, TelemetryClient? telemetryClient = null)
```

Comment:
Will only be null if the consumer application hasn't registered Application Insights services in DI